tests/qos/test_tunnel_qos_remap_nvidia.py

```diff
@@ -0,0 +1,231 @@
+        mg_facts = rand_selected_dut.get_extended_minigraph_facts(tbinfo)
+        active_tor_mac = rand_selected_dut.facts['router_mac']
+        uplink_interfaces = {}
+        t1_ptf_ports = []
+        for portchannel in mg_facts['minigraph_portchannels'].values():
+            for member in portchannel['members']:
+                ptf_indice = mg_facts['minigraph_ptf_indices'][member]
+                t1_ptf_ports.append(ptf_indice)
+                uplink_interfaces[ptf_indice] = member
+        # Always select the last port in the last LAG as src_port
+        last_t1_ptf_port = sorted(t1_ptf_ports)[-1]
+        # Randomly select a server to receive the traffic
+        vlan_interface = list(mg_facts['minigraph_vlans'].keys())[0]
+        downlink_interface = random.choice(mg_facts['minigraph_vlans'][vlan_interface]['members'])
+        server = mg_facts['minigraph_neighbors'][downlink_interface]['name']
+        server_ip = mg_facts['minigraph_devices'][server]['lo_addr']
+        server_ptf_port = mg_facts['minigraph_ptf_indices'][downlink_interface]
+
+    def _generate_packet(eth_dst, ip_src, ip_dst, ip_dscp):
+        pkt = simple_tcp_packet(
+            eth_dst=eth_dst,
+            ip_src=ip_src,
+            ip_dst=ip_dst,
+            ip_dscp=ip_dscp,
+            ip_ecn=1,
+            ip_ttl=64
+        )
+        exp_pkt = Mask(pkt)
+        exp_pkt.set_do_not_care_scapy(scapy.Ether, "dst")
+        exp_pkt.set_do_not_care_scapy(scapy.Ether, "src")
+        exp_pkt.set_do_not_care_scapy(scapy.IP, "id")  # since src and dst changed, ID would change too
+        exp_pkt.set_do_not_care_scapy(scapy.IP, "ttl")  # ttl in outer packet is kept default (64)
+        exp_pkt.set_do_not_care_scapy(
+            scapy.IP, "chksum")  # checksum would differ as the IP header is not the same
+        return pkt, exp_pkt
+
+    with allure.step("Start the uplink test"):
+        for dscp, queue in up_link_test_data:
+            pkt, expected_pkt = _generate_packet(active_tor_mac, '1.1.1.1', server_ip, dscp)
+            # Clear queue counters before sending traffic
+            rand_selected_dut.shell('sonic-clear queuecounters')
+            logger.info("Sending normal traffic from t1 to server {} with dscp {}".format(server_ip, dscp))
+            logger.info("The expected queue for the traffic is {}".format(queue))
+            testutils.send(ptfadapter, last_t1_ptf_port, pkt, PKT_NUM)
+            testutils.verify_packet(ptfadapter, expected_pkt, server_ptf_port)
+            pytest_assert(wait_until(
+                5, 1, 0, check_queue_counter, rand_selected_dut, [downlink_interface], queue, PKT_NUM),
+                "The queue counter for DSCP: {}, Queue: {} is not as expected: >= {}. "
+                "Please check the output of cmd 'show queue counters' in the test log.".format(dscp, queue, PKT_NUM))
+
+    with allure.step("Start the downlink test"):
```

Comment:
the downlink shares similar steps as the uplink, maybe we could reuse the code here.